### PR TITLE
Claim status detail screen

### PR DESCRIPTION
**Pull request by SatishBV**
## [APP-1135]

- Let's member see details of the claim status detail screen

## Checklist

- [x] Dark/light mode verification
- [x] Landscape verification
- [x] iPad verification
- [ ] iPod/iPhone 12 mini/iPhone SE verification

[APP-1135]: https://hedvig.atlassian.net/browse/APP-1135?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Projects/Home/Sources/Claims/ClaimDetailView.swift [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            case .paid = claim.claimDetailData.outcome,
+            !claim.claimDetailData.payout.amount.isEmpty
+        {
+            return claim.claimDetailData.payout
+        }
+
+        return nil
+    }
+
+    private var contactChatView: some View {
```
Comment (sampettersson) [inline Projects/Home/Sources/Claims/ClaimDetailView.swift:36]: I think this is a sign that this should be split into its own struct, right? 😄

File: Projects/Home/Sources/Claims/ClaimDetailHeader.swift [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    private var closedDate: String {
+        let dateFormatter = DateFormatter.withIso8601Format("yyyy-MM-dd'T'HH:mm:ss.SSSSSSZ")
+        guard let closed = closed,
+            let date = dateFormatter.date(from: closed)
+        else { return "-" }
+        return readableDateString(from: date)
+    }
+
+    /// Converts date into a readable friendly string
+    private func readableDateString(from date: Date) -> String {
```
Comment (sampettersson) [inline Projects/Home/Sources/Claims/ClaimDetailHeader.swift:44]: I remember that we have some very similar code in hCore, maybe we can reuse that?

File: Projects/Home/Sources/Claims/ClaimDetailHeader.swift [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    private var submittedDate: String {
+        let dateFormatter = DateFormatter.withIso8601Format("yyyy-MM-dd'T'HH:mm:ss.SSSSSSZ")
+        guard let submitted = submitted,
+            let date = dateFormatter.date(from: submitted)
+        else { return "-" }
+        return readableDateString(from: date)
+    }
+
+    private var closedDate: String {
+        let dateFormatter = DateFormatter.withIso8601Format("yyyy-MM-dd'T'HH:mm:ss.SSSSSSZ")
```
Comment (sampettersson) [inline Projects/Home/Sources/Claims/ClaimDetailHeader.swift:36]: Can we extend `String.localDateToDate` maybe?

File: Projects/Home/Sources/AudioTrack/TrackPlayer.swift [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    private func startPlaying() {
+        let session = AVAudioSession.sharedInstance()
+
+        do {
+            try session.setCategory(.playback)
+            try session.overrideOutputAudioPort(AVAudioSession.PortOverride.speaker)
+            try session.setActive(true)
+        } catch {
+            print("Playing over the device's speakers failed")
```
Comment (sampettersson) [inline Projects/Home/Sources/AudioTrack/TrackPlayer.swift:100]: Should we show an error to the user?

**Review by sampettersson — CHANGES_REQUESTED** [review]
Haven't checked on the phone yet but code looks super solid! Great job! Only thing is look through and try to split up the components a bit more, some of them are a bit chunky and hard to read :) also try to avoid multiple structs/classes in the same file.

File: Projects/Home/Sources/Claims/ClaimDetailHeader.swift (hunk 2 above)
Comment (SatishBV) [reply Projects/Home/Sources/Claims/ClaimDetailHeader.swift:44]: I needed to use RelativeDateTimeFormatter for this particular problem. I haven't found it to be used else where in the app. Do you know of any particular name by which I can find similar code?

If not, I can put this code in hCore

File: Projects/Home/Sources/AudioTrack/TrackPlayer.swift (hunk 4 above)
Comment (SatishBV) [reply Projects/Home/Sources/AudioTrack/TrackPlayer.swift]: I copied this code from Embark flow to initialise audio player. Chances of error here should be less.

But if an error  has to be shown, do you have any standard template code to show errors? Like should it be an alert or some other view?

File: Projects/App/Sources/Journeys/ClaimsJourney.swift [hunk 5]
```diff
@@ -96,5 +96,11 @@ extension AppJourney {
         ) { action in
             DismissJourney()
         }
+        .addConfiguration { presenter in
+            presenter.viewController.navigationItem.largeTitleDisplayMode = .never
+            presenter.viewController.navigationItem.title = L10n.ClaimStatus.title
```
Comment (sampettersson) [inline Projects/App/Sources/Journeys/ClaimsJourney.swift:101]: Title has a method `configureTitle` which you can use.

File: Projects/App/Sources/Journeys/ClaimsJourney.swift [hunk 6]
```diff
@@ -96,5 +96,11 @@ extension AppJourney {
         ) { action in
             DismissJourney()
         }
+        .addConfiguration { presenter in
+            presenter.viewController.navigationItem.largeTitleDisplayMode = .never
```
Comment (sampettersson) [inline Projects/App/Sources/Journeys/ClaimsJourney.swift:100]: Maybe good to break this out into a function and put it in hCore?

File: Projects/Home/Sources/Claims/ClaimDetailView.swift [hunk 7]
```diff
@@ -79,19 +79,19 @@ public struct ClaimDetailView: View {
 
             Spacer()
         }
-        .navigationBarTitle(Text(L10n.ClaimStatus.title), displayMode: .inline)
         .onReceive(
-            store.stateSignal
-                .plain()
-                .distinct()
+            store
```
Comment (sampettersson) [inline Projects/Home/Sources/Claims/ClaimDetailView.swift]: I think this logic can be solved neater with the `PresentableStoreLens`, check here for reference: https://github.com/HedvigInsurance/ugglan/blob/666c8bc60596482e28b1db53767dcb1ef77edfeb/Projects/Contracts/Sources/View/ContractTable.swift#L35

File: Projects/hGraphQL/Sources/Models/Claims.swift [hunk 8]
```diff
@@ -200,3 +200,20 @@ public struct ClaimData {
         claims = cardData.claimsStatusCards.map { .init(cardData: $0) }
     }
 }
+
+extension Claim: Equatable {
+    public static func ==(lhs: Claim, rhs: Claim) -> Bool {
+        return lhs.id == rhs.id
```
Comment (sampettersson) [inline Projects/hGraphQL/Sources/Models/Claims.swift:206]: Why not make the whole object equatable, why just check the id? Doing this usually leads to weird behaviours I've seen 😬

File: Projects/Home/Sources/Claims/ClaimDetailView.swift [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+public struct ClaimDetailView: View {
+    @State var claim: Claim
+    var store: HomeStore
+
+    public init(
+        claim: Claim
+    ) {
+        self.claim = claim
+        let store: HomeStore = globalPresentableStoreContainer.get()
```
Comment (sampettersson) [inline Projects/Home/Sources/Claims/ClaimDetailView.swift:15]: Why not use @PresentableStore to fetch this?

**Review by sampettersson — CHANGES_REQUESTED** [review]

**Review by sampettersson — APPROVED** [review]

**Pull request merged**